### DISCUSSION TITLE
Hydration node mismatch on undefined in slot of dynamic component

### DISCUSSION THREAD

**Discussion by m0ksem** [nuxt/nuxt#17856] [Questions]
Reproduction: https://codesandbox.io/s/nuxt-undefined-dynamic-slot-tp45e9?file=/components/Test.vue

I don't know why some-times codesendbox shows me error "/" path is undefined.

But it's not a main issue. If you try to reproduce this in local, you will see this warning:
![image](https://user-images.githubusercontent.com/23530004/157502255-833a71ec-b6d6-4645-8b7c-3616c9e648c2.png)

Am I missed something from Vue/nuxt docs, or it's a bug?

My real issue is that my Icon component can have something in slot or have a class. So every time I have undefined in slot or "" it will show me warning.

**Comment by lihanspace** [comment]
I also encountered this problem

**Comment by m0ksem** [comment]
Temporal fix for me is:

```vue
<template>
  <component :is="tag">
    <slot>
      <template v-if="icon">
         {{ icon }}
      </template>
    </slot>
  </component>
</template>
```
